### ISSUE TITLE
no-unknown-classes reports every class as unknown when the entry point uses Tailwind v4 prefix(...)

### ISSUE THREAD

**Issue by beckerei** [sergioazoc/oxlint-tailwindcss#29]
## Summary

When the Tailwind v4 entry point declares a prefix via `prefix(...)`, `oxlint-tailwindcss` fails to resolve the design system entirely: `tailwindcss/no-unknown-classes` then flags **every** class — including core utilities like `flex` / `items-center` — as "not a valid Tailwind class". Removing the prefix makes the plugin work correctly. This blocks adoption on any codebase that uses a Tailwind prefix.

## Versions

- `oxlint-tailwindcss`: 1.1.0
- `oxlint`: 1.68.0
- `tailwindcss`: 4.3.0
- Node: 24.x

## Minimal reproduction

**`styles.css`** (in a package where `@import "tailwindcss"` resolves):
```css
@import "tailwindcss" prefix(c);
```

**`.oxlintrc.json`**:
```json
{
  "jsPlugins": ["oxlint-tailwindcss"],
  "settings": { "tailwindcss": { "entryPoint": "styles.css" } },
  "rules": { "tailwindcss/no-unknown-classes": "error" }
}
```

**`probe.tsx`**:
```tsx
export const x = () => <div className="flex c:flex items-center c:items-center" />;
```

Run `oxlint`.

### Actual
All four classes are reported as unknown:
```
error tailwindcss(no-unknown-classes): "flex" is not a valid Tailwind class. Did you mean "peer"?
error tailwindcss(no-unknown-classes): "c:flex" is not a valid Tailwind class.
error tailwindcss(no-unknown-classes): "items-center" is not a valid Tailwind class.
error tailwindcss(no-unknown-classes): "c:items-center" is not a valid Tailwind class.
```
The "Did you mean peer" suggestion implies the resolved design system is nearly empty.

### Expected
With `prefix(c)`, prefixed classes (`c:flex`, `c:items-center`) should be recognized as valid, and unprefixed ones rejected.

## Control (works without prefix)

Identical setup but `styles.css` = `@import "tailwindcss";` → **0 errors**; the plugin resolves the design system correctly. So the failure is specific to `prefix(...)` handling.

## Notes

- Reproduced with both the combined form (`@import "tailwindcss" prefix(c)`) and the split form (`@import "tailwindcss/theme.css" ... prefix(c)` + `@import "tailwindcss/utilities.css"`).
- `settings.tailwindcss.entryPoint` is set and resolvable (the no-prefix control proves resolution works from the same path).

**Comment by sergioazoc** [comment]
Fixed in **v1.2.0** — thanks for the detailed report, the reproduction made this easy to track down. 🙏

**Root cause:** the precompute step read class names via `getClassList()` (which returns them *without* the prefix, e.g. `flex`) and validated them with `candidatesToCss()`. Under `prefix(c)`, only the *prefixed* form (`c:flex`) produces CSS, so every class validated as `null` and the design system precomputed empty — which is why `no-unknown-classes` flagged everything (and the suggestion pool was nearly empty, hence the "Did you mean peer").

**The fix:** the precompute now reads the prefix from the design system, applies it only when calling the Tailwind APIs, and stores everything prefix-free. `no-unknown-classes` is now prefix-aware:

- `c:flex`, `c:items-center` → **valid**
- `flex`, `items-center` (a Tailwind utility without the required prefix) → reported with a new `missingPrefix` message and a quick-fix that adds it (`flex` → `c:flex`)
- component classes from `@layer components` carry no prefix and stay valid

The prefix is detected automatically from your `entryPoint` — nothing extra to configure. The combined (`prefix(c)`) and split (`theme.css … prefix(c)` + `utilities.css`) forms are both covered.

`consistent-variant-order` and `enforce-sort-order` (strict mode) were also made prefix-aware so the prefix always stays in first position.

Please upgrade to `oxlint-tailwindcss@1.2.0` and let me know if anything still looks off. Closing as resolved.
